### ISSUE TITLE
Unable to load the public suffix list rules

### ISSUE THREAD

**Issue by madika** [Payum/Payum#734]
Getting this exception with Payum 1.5 and Symfony 3.4

> Exceptions
> Unable to load the public suffix list rules for https://publicsuffix.org/list/public_suffix_list.dat

Payum 1.4 with Symfony 3.4 working normally

**Comment by makasim** [comment]
A stack trace would be great.

**Comment by madika** [comment]
```
League\Uri\PublicSuffix\Exception:
Unable to load the public suffix list rules for https://publicsuffix.org/list/public_suffix_list.dat

  at vendor/league/uri-hostname-parser/src/PublicSuffix/ICANNSectionManager.php:70
  at League\Uri\PublicSuffix\ICANNSectionManager->getRules('https://publicsuffix.org/list/public_suffix_list.dat')
     (vendor/league/uri-hostname-parser/src/functions.php:40)
  at League\Uri\resolve_domain('127.0.0.1')
     (vendor/league/uri-components/src/Components/Host.php:253)
  at League\Uri\Components\Host->getDomainInfo()
     (vendor/league/uri-components/src/Components/Host.php:174)
  at League\Uri\Components\Host->__construct('127.0.0.1')
     (vendor/league/uri-manipulations/src/Modifiers/UriMiddlewareTrait.php:142)
  at League\Uri\Modifiers\HostToAscii->filterHost('127.0.0.1')
     (vendor/league/uri-manipulations/src/Modifiers/HostToAscii.php:39)
  at League\Uri\Modifiers\HostToAscii->modifyHost('127.0.0.1')
     (vendor/league/uri-manipulations/src/Modifiers/HostMiddlewareTrait.php:31)
  at League\Uri\Modifiers\HostToAscii->execute(object(Http))
     (vendor/league/uri-manipulations/src/Modifiers/UriMiddlewareTrait.php:72)
  at League\Uri\Modifiers\HostToAscii->process(object(Http))
     (vendor/league/uri-manipulations/src/Modifiers/Pipeline.php:69)
  at League\Uri\Modifiers\Pipeline->League\Uri\Modifiers\{closure}(object(Http), object(HostToAscii))
  at array_reduce(array(object(HostToAscii), object(KsortQuery), object(DecodeUnreservedCharacters), object(RemoveDotSegments)), object(Closure), object(Http))
     (vendor/league/uri-manipulations/src/Modifiers/Pipeline.php:72)
  at League\Uri\Modifiers\Pipeline->execute(object(Http))
     (vendor/league/uri-manipulations/src/Modifiers/UriMiddlewareTrait.php:72)
  at League\Uri\Modifiers\Pipeline->process(object(Http))
     (vendor/league/uri-manipulations/src/Modifiers/Normalize.php:38)
  at League\Uri\Modifiers\Normalize->execute(object(Http))
     (vendor/league/uri-manipulations/src/Modifiers/UriMiddlewareTrait.php:72)
  at League\Uri\Modifiers\Normalize->process(object(Http))
     (vendor/payum/payum/src/Payum/Core/Security/Util/RequestTokenVerifier.php:21)
  at Payum\Core\Security\Util\RequestTokenVerifier::isValid('http://127.0.0.1:21000/payment/capture/bjIZbtefQeBbIsT1sVif80hCPRNmAzmBFrcMWR96Zyw', 'http://127.0.0.1:21000/payment/capture/bjIZbtefQeBbIsT1sVif80hCPRNmAzmBFrcMWR96Zyw')
     (vendor/payum/payum/src/Payum/Core/Bridge/Symfony/Security/HttpRequestVerifier.php:52)
  at Payum\Core\Bridge\Symfony\Security\HttpRequestVerifier->verify(object(Request))
     (vendor/payum/payum-bundle/Controller/CaptureController.php:32)
  at Payum\Bundle\PayumBundle\Controller\CaptureController->doAction(object(Request))
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:151)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:68)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:202)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (web/app_dev.php:45)
```

**Comment by makasim** [comment]
I suspect league url does not like this '127.0.0.1'. Could you try `http://localhost` or `http://127.0.01`?

**Comment by madika** [comment]
I'm using the Payum-Bundle, which does parsing/verifying the Symfony Http Request automatically.

I checked the source and it seems like the RequestTokenVerifier is automatically removing the http from the request uri.

**Comment by makasim** [comment]
should be fixed then.

**Comment by madika** [comment]
I just checked deeper and it seems like the CURL request to the PSL list is downloaded regulary, but it fails at storing the list/writing the cache.

I think it is wrong where it stores the cache file. I just tried to chmod 777 the whole folder and saw it stored the cache file under /vendor/league/uri-hostname-parser/data

This doesn't seem right, but I don't know if you can configure the hostname parser to use a different cache folder.

**Comment by garak** [comment]
I just encountered this error. I'm also using Symfony 3.4 and Payum 1.5.0 (PayumBundle 2.3.0)

**Comment by garak** [comment]
So, it looks like uri-components added an option to customize cache, so maybe now it would be possible to read/write suche cache in a writable dir? Maybe in sys_get_temp_dir() for this library and inside Symfony cache for the bundle?
See https://github.com/thephpleague/uri-components/pull/11 for a reference

**Comment by stefandoorn** [comment]
@garak I think that would be a proper solution. Unfortunately I don't know anything about the Payum internals, so not really easy to contribute for me.

**Comment by Great-Antique** [comment]
https://github.com/thephpleague/uri-components/pull/12
After `composer require league/uri-components:^1.7.1` everything works fine

**Comment by Tetragramat** [comment]
If it is fixed in league/uri-components:^1.7.1, then this could be closed.
Wouldn't be bad idea to add conflict with league/uri-components: <1.7.1 ?

**Comment by makasim** [comment]
Could you work on a PR?
